src/Moq/Interception/InterceptionAspects.cs

```diff
@@ -331,4 +332,131 @@ public static void Handle(Invocation invocation, Mock mock)
 			}
 		}
 	}
+
+	internal static class HandleAutoSetupProperties
+	{
+		private static readonly int AccessorPrefixLength = "?et_".Length; // get_ or set_
+
+		public static void Handle(Invocation invocation, Mock mock)
+		{
+			MethodInfo invocationMethod = invocation.Method;
+			// Original implementation with eager setup skipped indexers too.
+			if (invocationMethod.IsPropertyAccessor() && !invocationMethod.IsPropertyIndexerAccessor())
+			{
+				if (mock.Setups.FindMatchFor(invocation) == null)
+				{
+					PropertyInfo property = GetPropertyFromAccessorMethod(mock, invocationMethod, out Type propertyHolderType);
+
+					// Should ignore write-only properties, as they will be handled by Return aspect.
```

Comment:
... unless if the mock is strict; for those, interception terminates even before we get to `Return`!